tests/PHPUnit/Builder/KataBuilderTest.php

```diff
@@ -0,0 +1,109 @@
+<?php
+/**
+ * This file is part of the phpkata project.
+ *
+ * (c) Yannick Voyer (http://github.com/yvoyer)
+ */
+
+namespace Star\Kata\Builder;
+
+use Star\Kata\Model\Kata;
+
+/**
+ * Class KataBuilderTest
+ *
+ * @author  Yannick Voyer (http://github.com/yvoyer)
+ *
+ * @package Star\Kata\Builder
+ *
+ * @covers Star\Kata\Builder\KataBuilder
+ * @uses Star\Kata\Builder\ObjectiveBuilder
+ * @uses Star\Kata\Model\Objective\ConfigurableObjective
+ * @uses Star\Kata\Model\Kata
+ */
+final class KataBuilderTest extends \PHPUnit_Framework_TestCase
+{
+    /**
+     * @var KataBuilder
+     */
+    private $builder;
+
+    public function setUp()
+    {
+        $this->builder = new KataBuilder();
+    }
+
+    public function test_build_should_always_return_a_kata()
+    {
+        $this->assertInstanceOfKata($this->builder->build());
+    }
+
+    /**
+     * @depends test_build_should_always_return_a_kata
+     */
+    public function test_should_return_a_kata_with_default_name()
+    {
+        $kata = $this->builder->build();
+        $this->assertSame('kata-1', $kata->getName());
+    }
+
+    /**
+     * @depends test_build_should_always_return_a_kata
+     */
+    public function test_should_return_a_different_instance_on_successive_build()
+    {
+        $this->assertNotSame($this->builder->build(), $this->builder->build(), 'Should not be same instance');
+        $this->assertNotEquals($this->builder->build(), $this->builder->build(), 'Should not be same entity');
+    }
+
+    /**
+     * @depends test_build_should_always_return_a_kata
+     */
+    public function test_should_change_name_of_kata_on_each_build()
+    {
+        $this->assertSame('kata-1', $this->builder->build()->getName());
+        $this->assertSame('kata-2', $this->builder->build()->getName());
+        $this->assertSame('kata-3', $this->builder->build()->getName());
+    }
+
+    /**
+     * @depends test_build_should_always_return_a_kata
+     */
+    public function test_should_build_kata_with_name()
+    {
+        $kata = $this->builder
+            ->withName('my-name')
+            ->build();
+
+        $this->assertSame('my-name', $kata->getName());
```

Comment:
I would have implemented a uniqueid name when no name is given. So for me its legit, since i did not specified any name. A kata2 name seems more appropriate to me than '' (empty string). Unless you have another advice.